deploy/crd.yaml

```diff
@@ -11,9 +11,357 @@ spec:
     singular: flinkapplication
     shortNames:
       - flinkapp
-
   scope: Namespaced
   version: v1alpha1
+  validation:
+    # openAPIV3Schema is the schema for validating custom objects.
+    openAPIV3Schema:
+      type: object
+      properties:
+        spec:
+          type: object
+          properties:
+            image:
+              type: string
+            imagePullPolicy:
+              type: string
+              enum: [Always, Never, IfNotPresent]
+            imagePullSecrets:
+              type: array
+              items:
+                type: object
+                properties:
+                  name:
+                    type: string
+            jarName:
+              type: string
+            programArgs:
+              type: string
+            entryClass:
+              type: string
+            flinkVersion:
+              type: string
+            restartNonce:
+              type: string
+            parallelism:
+              type: integer
+              minimum: 1
+            deleteMode:
+              type: string
+              enum: [Savepoint, None, ForceCancel]
+            deploymentMode:
+              type: string
+              enum: [Single, Dual]
+            rpcPort:
+              type: integer
+              minimum: 1
+              maximum: 65535
+            blobPort:
+              type: integer
+              minimum: 1
+              maximum: 65535
+            queryPort:
+              type: integer
+              minimum: 1
+              maximum: 65535
+            metricsQueryPort:
+              type: integer
+              minimum: 1
+              maximum: 65535
+            flinkConfig:
+              type: object
+              properties:
+                additionalProperties:
+                  type: string
```

Comment:
I think this is too restrictive. In the go definition this is `map[string]interface{}`, and there are flink configs that are numeric and boolean in addition to string.

Reply:
The problem with `map[string]interface{}` is that it can allow nested structure. On the other end we are appending only strings  to the conf.yaml.

I tested with integer values like `taskmanager.heap.size: 200` and boolean values and they work. It integer as numeric string and does not fail.